connect-rest-extension/src/main/java/io/debezium/kcdrestextension/DebeziumResource.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.kcdrestextension;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.connect.health.ConnectClusterState;
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.health.ConnectClusterStateImpl;
+import org.apache.kafka.connect.runtime.isolation.PluginDesc;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.predicates.HasHeaderKey;
+import org.apache.kafka.connect.transforms.predicates.RecordIsTombstone;
+import org.apache.kafka.connect.transforms.predicates.TopicNameMatches;
+import org.apache.maven.artifact.versioning.ComparableVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.kcdrestextension.entities.TransformsInfo;
+
+@Path("/debezium")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class DebeziumResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebeziumResource.class);
+
+    // TODO: This should not be so long. However, due to potentially long rebalances that may have to wait a full
+    // session timeout to complete, during which we cannot serve some requests. Ideally we could reduce this, but
+    // we need to consider all possible scenarios this could fail. It might be ok to fail with a timeout in rare cases,
+    // but currently a worker simply leaving the group can take this long as well.
+    public static final long REQUEST_TIMEOUT_MS = 90 * 1000;
+    // Mutable for integration testing; otherwise, some tests would take at least REQUEST_TIMEOUT_MS
+    // to run
+    private static long requestTimeoutMs = REQUEST_TIMEOUT_MS;
+
+    private final List<TransformsInfo> transforms;
+    private Boolean isTopicCreationEnabled = null;
+    private final Herder herder;
+    private final Map<String, ?> config;
+
+    private static final ComparableVersion TOPIC_CREATION_KAFKA_VERSION = new ComparableVersion("2.6.0");
+
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public DebeziumResource(ConnectClusterState clusterState, Map<String, ?> config) {
+        Field herderField;
+        try {
+            herderField = ConnectClusterStateImpl.class.getDeclaredField("herder");
```

Comment:
Have you started a discussion with the upstream Kafka community, exploring whether we can get "official" access to this?

Reply:
I prepared for it but did not kick it off yet.